### PR TITLE
Change the way we run e2e tests with feature gates in pipelines

### PR DESCRIPTION
**Pull request by sbwsg**
# Changes

Prior to this commit pipelines ran e2e tests for both "stable" and
"alpha" feature gates in a single prow job. This makes the tests
take twice as long as running running either individually.

In a [parallel commit to Plumbing](https://github.com/tektoncd/plumbing/pull/833) we are introducing a new prow job
that is specifically for running alpha e2e tests. This means both
the alpha and stable tests can be run concurrently and hopefully
should reduce the iteration time on PRs - failures should show up
sooner.

The alpha feature gate prow job will set the PIPELINE_FEATURE_GATE
env var when it is initiated. Our e2e script recognizes this var
and sets the feature-flags configmap accordingly. If the env var
is not set then we default to running under "stable" feature gate.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by sbwsg** [comment]
Let's see if this works so soon after the plumbing merge:

/test pull-tekton-pipeline-alpha-integration-tests

**Comment by sbwsg** [comment]
Re-running integration too to see if that now skips alpha:

/test pull-tekton-pipeline-integration-tests

**Comment by sbwsg** [comment]
It looks like a timeout test failed because of cluster slowness. Let's rerun and see what happens.

/test pull-tekton-pipeline-alpha-integration-tests

**Comment by sbwsg** [comment]
nice.  everything's passed and the integration runs each set their own feature gates as expected.

**Review by vdemeester — APPROVED** [review]

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -52,11 +52,14 @@ function run_e2e() {
   go_test_e2e -tags=examples -timeout=20m ./test/ || failed=1
 }
 
-set_feature_gate "stable"
-run_e2e
+if [ "$PIPELINE_FEATURE_GATE" == "" ]; then
```
Comment (afrittoli) [inline test/e2e-tests.sh:55]: NIT: We could also do: 
```shell
set_feature_gate "${PIPELINE_FEATURE_GATE:-stable}"
run_e2e
```

But this works too.

**Review by afrittoli — COMMENTED** [review]
Thank you!!
/lgtm

**Pull request merged**